Util/openl-opentelemetry-javaagent-extension/README.md

```diff
@@ -0,0 +1,126 @@
+which can affect performance. Therefore, it is recommended to enable open method logging only to find methods causing performance problems.
+
+## For developers:
+
+## Setup
+
+Download and setup opentelemetry javaagent as described in [opentelemetry documentation](https://opentelemetry.io/docs/instrumentation/java/automatic/#setup).
+Or use `opentelemetry-javaagent.jar` file downloaded to `\target` folder when the current project was built.
+Run jaeger to receive metrics.
+
+## Configuring the agent
+
+Add next configuration to webservice app VM options:
+
+```
+-javaagent:<path/to>/opentelemetry-javaagent.jar
+-Dotel.traces.exporter=jaeger
+-Dotel.exporter.jaeger.endpoint=http://localhost:14250
+-Dotel.propagators=jaeger
+-Dotel.metrics.exporter=none
+-Dotel.service.name=RuleServices
+```
+
+or with environment variables:
+
+```
+JAVA_TOOL_OPTIONS=-javaagent:<path/to>/opentelemetry-javaagent.jar
+OTEL_TRACES_EXPORTER=jaeger
+OTEL_EXPORTER_JAEGER_ENDPOINT=http://localhost:14250
+OTEL_PROPAGATORS=jaeger
+OTEL_METRICS_EXPORTER=none
+OTEL_SERVICE_NAME=RuleServices
+```
+
+Agent configured to send logs to Jaeger exporter. `otel.tracer.exporter` value should be set to actual Jaeger address. 
+The default value for metrics exporter is `otlp`, and there are no metrics for jaeger in opentelemetry, so it is set to `none` value.
+`otel.service.name` value can be changed at will.
+
+This configuration will log all http requests, receiving/sending data via kafka and others known libraries and frameworks presented in OpenL project. 
+To suppress some undesired agent instrumentation you can add `-Dotel.instrumentation.[name].enabled=false` to the configuration where `name` is the corresponding instrumentation name. 
+See [Suppressing specific agent instrumentation](https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#suppressing-specific-agent-instrumentation).
+
+## Configuring the agent for OpenL methods 
+
+By default, opentelemetry javaagent will send logs from most known libraries and frameworks. But OpenL is not in the list.
+To enable logging of OpenL methods, you need to include these methods to configuration.
+
+### Adding OpenL methods to tracing
+
+To add OpenL methods for tracing add next property to opentelemetry javaagent configuration:
+
+```
+-Dotel.instrumentation.methods.include=org.openl.rules.calc.Spreadsheet[innerInvoke];org.openl.rules.dt.DecisionTable[innerInvoke];org.openl.rules.tbasic.Algorithm[innerInvoke];org.openl.rules.method.table.TableMethod[innerInvoke];org.openl.rules.cmatch.ColumnMatch[innerInvoke]
+```
+
+or with environment variables:
+
+```
+OTEL_INSTRUMENTATION_METHODS_INCLUDE=org.openl.rules.calc.Spreadsheet[innerInvoke];org.openl.rules.dt.DecisionTable[innerInvoke];org.openl.rules.tbasic.Algorithm[innerInvoke];org.openl.rules.method.table.TableMethod[innerInvoke];org.openl.rules.cmatch.ColumnMatch[innerInvoke]
+```
+
+It is recommended to add classes and methods from the list above.
```

Comment:
Can we hide this configuration? It is magic for the user and it is very easy to make there error.

Reply:
Yes, I'll fix it